### PR TITLE
limit order

### PR DESCRIPTION
**Pull request by efang05**
renamed market order to limit order then added actual market orders

### CODE REVIEW

**Review by stevenewald — CHANGES_REQUESTED** [review]
Can we add some tests as well?

File: linter/src/pywrapper/runtime.hpp [hunk 1]
```diff
@@ -12,7 +12,7 @@ namespace nutc {
 namespace pywrapper {
 [[nodiscard]] bool create_api_module(
     std::function<bool(const std::string&, const std::string&, float, float)>
-        publish_market_order
+        publish_limit_order
```
Comment (glasss13) [inline linter/src/pywrapper/runtime.hpp:15]: To keep the code consistnent, we should probably make this function take in both `publish_limit_order` and `publish_market_order` like how it is in `exchange`

File: exchange/test/test_algos/confirm_market_order.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
+def place_limit_order(side: str, ticker: str, quantity: float, price: float) -> None:
+    """Place a limit order - DO NOT MODIFY"""
+
+def place_market_order(side: str, ticker: str, quantity: float) -> None:
+    """Place a market order - DO NOT MODIFY"""
```
Comment (glasss13) [inline exchange/test/test_algos/confirm_market_order.py:5]: We should probably add this to the actual template as well.

**Comment by stevenewald** [comment]
Also pls merge with main @efang05

File: exchange/test/src/integration/basic.cpp [hunk 3]
```diff
@@ -43,6 +43,37 @@ TEST_F(IntegrationBasicAlgo, InitialLiquidity)
...
+    auto trader2 = users_.add_trader<TestTrader>(0);
+    trader2->modify_holdings("TSLA", 1000); // NOLINT
+    trader2->add_order({sell, "TSLA", 100, 1000});
+
+    TestMatchingCycle cycle{
+        {"TSLA"},
+        {trader1, trader2},
+    };
+
+    cycle.wait_for_order({buy, "APPL", 1, 1000});
```
Comment (stevenewald) [inline exchange/test/src/integration/basic.cpp:58]: Shouldn't we be waiting for the market order? Why does buy_market_order place a limit order?

**Comment by glasss13** [comment]
I think you accidentally included RSI bot into this PR

**Comment by stevenewald** [comment]
I'm gonna wrap this up because I made some changes that broke it, good work @efang05

**Pull request merged**